diff --git a/web/apps/labelstudio/src/components/Menubar/Menubar.jsx b/web/apps/labelstudio/src/components/Menubar/Menubar.jsx
index 3dafe0e..5c6b3de 100644
--- a/web/apps/labelstudio/src/components/Menubar/Menubar.jsx
+++ b/web/apps/labelstudio/src/components/Menubar/Menubar.jsx
@@ -133,16 +133,18 @@ export const Menubar = ({ enabled, defaultOpened, defaultPinned, children, onSid
     useMenuRef?.current?.close();
   }, [location]);
 
+  const embed = window.location !== window.parent.location;
+
   return (
     <div className={contentClass}>
       {enabled && (
         <div className={menubarClass}>
-          <Dropdown.Trigger dropdown={menuDropdownRef} closeOnClickOutside={!sidebarPinned}>
+          {!embed && <Dropdown.Trigger dropdown={menuDropdownRef} closeOnClickOutside={!sidebarPinned}>
             <div className={`${menubarClass.elem("trigger")} main-menu-trigger`}>
               <LSLogo className={`${menubarClass.elem("logo")}`} alt="Label Studio Logo" />
               <Hamburger opened={sidebarOpened} />
             </div>
-          </Dropdown.Trigger>
+          </Dropdown.Trigger>}
 
           <div className={menubarContext}>
             <LeftContextMenu className={contextItem.mod({ left: true }).toClassName()} />
@@ -176,7 +178,7 @@ export const Menubar = ({ enabled, defaultOpened, defaultPinned, children, onSid
 
           {ff.isActive(ff.FF_THEME_TOGGLE) && <ThemeToggle />}
 
-          <Dropdown.Trigger
+          {!embed && <Dropdown.Trigger
             ref={useMenuRef}
             align="right"
             content={
@@ -207,7 +209,7 @@ export const Menubar = ({ enabled, defaultOpened, defaultPinned, children, onSid
               <Userpic user={user} isInProgress={isLoading} />
               {showNewsletterDot && <div className={menubarClass.elem("userpic-badge").toClassName()} />}
             </div>
-          </Dropdown.Trigger>
+          </Dropdown.Trigger>}
         </div>
       )}
 
