commit 9299d6581e558cd4d35031dde03b2cb810dfcc46
Author: Andraž Andolšek Peklaj <andraz.ap@pm.me>
Date:   Thu Mar 5 15:17:19 2026 +0100

    Bypass login

diff --git a/label_studio/users/views.py b/label_studio/users/views.py
index 404b8a99d..e819bb966 100644
--- a/label_studio/users/views.py
+++ b/label_studio/users/views.py
@@ -17,6 +17,7 @@ from organizations.models import Organization
 from rest_framework.authtoken.models import Token
 from users import forms
 from users.functions import login, proceed_registration
+from users.models import User
 
 logger = logging.getLogger()
 
@@ -104,6 +105,16 @@ def user_signup(request):
 def user_login(request):
     """Login page"""
     user = request.user
+
+    if not user.is_authenticated:
+        if User.objects.exists():
+            user = User.objects.first()
+        else:
+            user = User.objects.create_user("user@example.com", "hackme", allow_newsletters=False, username="User")
+            user.active_organization = Organization.create_organization(created_by=user, title='Label Studio')
+            user.save(update_fields=['active_organization'])
+        login(request, user, backend='django.contrib.auth.backends.ModelBackend')
+
     next_page = request.GET.get('next')
 
     # checks if the URL is a safe redirection.
