diff --git a/web/apps/labelstudio/src/components/HeidiTips/utils.ts b/web/apps/labelstudio/src/components/HeidiTips/utils.ts
index 4202fb4f7..7dfbb9948 100644
--- a/web/apps/labelstudio/src/components/HeidiTips/utils.ts
+++ b/web/apps/labelstudio/src/components/HeidiTips/utils.ts
@@ -42,55 +42,7 @@ export function getTipMetadata(tip: Tip) {
   };
 }
 
-export const loadLiveTipsCollection = () => {
-  // stale while revalidate - we will return the data present in the cache or the default data and fetch updated data to be put into the cache for the next time this function is called without waiting for the promise.
-  const cachedData = localStorage.getItem(CACHE_KEY);
-  const fetchedAt = localStorage.getItem(CACHE_FETCHED_AT_KEY);
-
-  // Read from local storage if the cachedData is less than CACHE_STALE_TIME milliseconds old
-  if (cachedData && fetchedAt && Date.now() - Number.parseInt(fetchedAt) < CACHE_STALE_TIME) {
-    return JSON.parse(cachedData);
-  }
-
-  const abortController = new AbortController();
-
-  // Abort the request after MAX_TIMEOUT milliseconds to ensure we won't wait for too long, something might be wrong with the network or it could be an air-gapped instance
-  const abortTimeout = setTimeout(abortController.abort, MAX_TIMEOUT);
-
-  // Fetch from github raw liveContent.json proxied through the server
-  fetch("/heidi-tips", {
-    headers: {
-      "Cache-Control": "no-cache",
-      "Content-Type": "application/json",
-    },
-    signal: abortController.signal,
-  })
-    .then(async (response) => {
-      if (response.ok) {
-        const data = await response.json();
-
-        // Cache the fetched content
-        localStorage.setItem(CACHE_FETCHED_AT_KEY, String(Date.now()));
-        localStorage.setItem(CACHE_KEY, JSON.stringify(data));
-      }
-    })
-    .catch((e) => {
-      console.warn("Failed to load live Heidi tips collection", e);
-    })
-    .finally(() => {
-      // Wait until the content is fetched to clear the abort timeout
-      // The abort should consider the entire request not just the headers
-      clearTimeout(abortTimeout);
-    });
-
-  // Serve possibly stale cached content
-  if (cachedData) {
-    return JSON.parse(cachedData);
-  }
-
-  // Default local content
-  return defaultTipsCollection;
-};
+export const loadLiveTipsCollection = () => { return {}; };
 
 export function getRandomTip(collection: keyof TipsCollection): Tip | null {
   const tipsCollection = loadLiveTipsCollection();
